controllers/osrelease.go

```diff
@@ -0,0 +1,81 @@
+package controllers
+
+import (
+	"github.com/ashcrow/osrelease"
+	"strings"
+)
+
+// OS Release Paths
+const (
+	EtcOSReleasePath string = "/etc/os-release"
+	LibOSReleasePath string = "/usr/lib/os-release"
+)
+
+// OS IDs
+const (
+	coreos string = "coreos"
+	fedora string = "fedora"
+	rhcos  string = "rhcos"
+	scos   string = "scos"
+)
+
+// OperatingSystem is a wrapper around a subset of the os-release fields
+// and also tracks whether ostree is in use.
+type OperatingSystem struct {
+	// id is the ID field from the os-release
+	id string
+	// variantID is the VARIANT_ID field from the os-release
+	variantID string
+	// version is the VERSION, RHEL_VERSION, or VERSION_ID field from the os-release
+	version string
+	// osrelease is the underlying struct from github.com/ashcrow/osrelease
```

Comment:
I was bit sceptic about using the osrelease project as it has not been updated since last 6 years. However found that MCO is also using  it in `machine-config-operator/pkg/daemon/osrelease.go`.  So should be ok :-)